trinity/plugins/eth2/beacon/plugin.py

```diff
@@ -178,7 +174,6 @@ def new_slot(self, slot: int) -> None:
         """
         The callback for `SlotTicker`, to be called whenever new slot is ticked.
         """
-        assert slot > self.eth2_config.GENESIS_SLOT
         head = self.chain.get_canonical_head()
         state_machine = self.chain.get_state_machine(at_block=head)
```

Comment:
If we have more than one fork, you're not building the new block on the new fork if you set `at_block=head`.

I *think* the correct logic is:
1. use `get_state_machine_class_for_block_slot` to get the `state_machine_class` of the *next* block.
2. generate a block from the parent with of the given `state_machine_class`.
3. use it to initiate a `StateMachine`.

So, could you try if it works?
```suggestion
        state_machine = self.chain.get_state_machine()
```
If `at_block` is `None`, it should do the above logic.